python/Ice/bidir/server/server.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+# Copyright (c) ZeroC, Inc.
+
+import asyncio
+import time
+import Ice
+from EarlyRiser import ButtonPressed, WakeUpService, AlarmClockPrx
+from common.time import toTimePoint
+
+
+class BidirWakeUpService(WakeUpService):
+
+    def __init__(self, communicator, eventLoop):
+        self._communicator = communicator
+        self._eventLoop = eventLoop
+
+    def wakeMeUp(self, timestamp, current):
+        timePoint = toTimePoint(timestamp)
+        print(f"Dispatching wakeMeUp request {{ timeStamp = '{timePoint.isoformat()}' }}")
+
+        connection = current.con # The connection from the client to the server.
+        if (connection is None):
+            # Should never happen, but in case it does, the Ice runtime transmits this exception as an
+            # Ice.UnknownException.
+            raise NotImplementedError("BidirWakeUpService does not support collocated calls")
+        
+        # Create a proxy to the client's alarm clock. This connection-bound proxy is called a "fixed proxy".
+        alarmClock = AlarmClockPrx.uncheckedCast(connection.createProxy(Ice.stringToIdentity("alarmClock")))
+        
+        self._eventLoop.call_soon_threadsafe(lambda: asyncio.create_task(self.callRing(alarmClock, timePoint)))
```

Comment:
This looks pretty complicated. There is not a simpler way to perform a "Task.Run"?

Reply:
What about:

```
asyncio.run_coroutine_threadsafe(self.callRing(alarmClock, timePoint), self._eventLoop)
```

Reply:
Better, but still pretty complicated.

Reply:
I guess we can also make, `wakeMeUp` an `async def` method so that it runs in the event loop. And then use `_eventLoop.call_soon(self.callRing(alarmClock, timePoint)`.